### DISCUSSION TITLE
-page not change dimension. Is this a bug?

### DISCUSSION THREAD

**Discussion by lmiol** [ImageMagick/ImageMagick#7530] [Help] [answered]
### ImageMagick version

7.1.1-36-portable-Q16-x64

### Operating system, version and so on

Windows 10

### Description

I have 3 files 
Platform1.png 836x709
Platform2.png 806x655
Lagoon.jpg 2048x1151

`magick  ( -label "top" Platform1.png -page 280x228+0+0 -background none ) ( -label "middle" Platform2.png -page 280x228+969+366 -background none  ) ( -label "background" Lagoon.jpg -page 2048x1151+0+0 -mosaic ) ( -clone 0--1 -mosaic ) -reverse "output.psd`"
after execute
top layer page geometry is 836x709+0+0 (dimensions the same as file)
middle layer page geometry is 806x655+969+366 (dimensions the same as file)
middle layer page geometry is 2048x1151+0+0  (dimensions the same as file)
so coordinates of X and Y are works well but dimensions is not. Is it a bug or i did wrong script?

**Comment by snibgo** [comment]
Your command shows different filenames from your description of "I have 3 files". Why is this?

Please write to `info:` instead of "`output.psd`". This will list each image, with the virtual canvas numbers. What is the text output?

Please also show the text outputs from:
```
magick Platform_Lagoon1A.png info:
```
... for each of ypur input files.

**Reply by lmiol** [reply]
> Your command shows different filenames from your description of "I have 3 files". Why is this?

my bad, i did update description 

> Please write to `info:` ... for each of ypur input files.

Lagoon.jpg JPEG 2048x1151 2048x1151+0+0 8-bit sRGB 1.58259MiB 0.000u 0:00.000
Platform1.png PNG 836x709 836x709+0+0 8-bit TrueColorAlpha sRGB 312494B 0.000u 0:00.000
Platform2.png PNG 806x655 806x655+0+0 8-bit TrueColorAlpha sRGB 340896B 0.000u 0:00.000

**Comment by snibgo** [comment]
And what is the text result when you write to `info:` instead of "`output.psd`"?
```
magick  ( -label "top" Platform1.png -page 280x228+0+0 -background none ) ( -label "middle" Platform2.png -page 280x228+969+366 -background none  ) ( -label "background" Lagoon.jpg -page 2048x1151+0+0 -mosaic ) ( -clone 0--1 -mosaic ) -reverse info:
```

**Reply by lmiol** [reply]
Platform1.png[0] PNG 2048x1151 2048x1151+0+0 8-bit sRGB 312494B 0.001u 0:00.001
Lagoon.jpg[1] JPEG 2048x1151 2048x1151+0+0 8-bit sRGB 1.58259MiB 0.008u 0:00.007
Platform2.png[2] PNG 806x655 280x228+969+366 8-bit TrueColorAlpha sRGB 340896B 0.014u 0:00.013
Platform1.png[3] PNG 836x709 280x228+0+0 8-bit TrueColorAlpha sRGB 312494B 0.029u 0:00.029

but in output.psd another data
i mean i checked by
`magick identify -verbose "output.psd" > output.txt`
here is log [output.txt](https://github.com/user-attachments/files/16500477/output.txt)

 Page geometry: 2048x1151+0+0
 Page geometry: 836x709+0+0
 Page geometry: 806x655+969+366

**Reply by fmw42** [reply]
Put -page before the image or use -set page after the image

**Reply by lmiol** [reply]
> Put -page before the image 

`magick  ( -label "top" -page 280x228+0+0 Platform1.png -background none ) ( -label "middle" -page 280x228+969+366 Platform2.png -background none  ) ( -label "background" -page 2048x1151+0+0 Lagoon.jpg -mosaic ) ( -clone 0--1 -mosaic ) -reverse output.psd`

> or use -set page after the image

`magick  ( -label "top"  Platform1.png -set page 280x228+0+0 -background none ) ( -label "middle" Platform2.png  -set page 280x228+969+366 -background none  ) ( -label "background" Lagoon.jpg -set page 2048x1151+0+0 -mosaic ) ( -clone 0--1 -mosaic ) -reverse output.psd`

both doesn't work

**Reply by fmw42** [reply]
How do you know they do not work.  What are you expecting?  Do you understand what -page and -set page are supposed to do?

**Comment by snibgo** [comment]
I'm not sure what result you want. In this command:
```
magick  
( -label "top" Platform1.png -page 280x228+0+0 -background none ) 
( -label "middle" Platform2.png -page 280x228+969+366 -background none  ) 
( -label "background" Lagoon.jpg -page 2048x1151+0+0 -mosaic ) 
( -clone 0--1 -mosaic ) 
-reverse info:
```
... you don't change the size of the "top" or "middle" images, so they remain at the size of the images in `Platform1.png` and `Platform2.png`.

`-page` doesn't change the dimensions of an image. If you want to change the size of the image, I suggest `-extent` or `-resize`, but there are other methods.

**Reply by lmiol** [reply]
page geometry is not rasterization of the image (-resize) but a kind of saving the new resolution taking into account the original resolution of the image.
For example, if you have an image of 2000 pixels, and you did -page geomerty 10x10+10+10 this means that when you open it in the editor and increase it back using free transform, for example, the image will be of good resolution again. In other words, the quality does not deteriorate.

-resize does the actual degradation of the image and you will not be able to increase it from 10px.

photopea.com is a free site and works with PSD, it also uses imagemagick to work with PSD.

Here is a file created in Photopea.
[output_photopea.zip](https://github.com/user-attachments/files/16502134/output_photopea.zip)

You can open via Photopea and increase the 280x220 layer
![image](https://github.com/user-attachments/assets/f75dc70f-1331-4450-ae4a-8575fe6be63b)

 to 1000x1000 size and it will not lose quality.
![image](https://github.com/user-attachments/assets/474ac827-d86e-4110-af52-d63f8d62f11f)

In this case, if you consider the data -verbose, the layer will have information like
page geomerty 280x228+969+366
i.e. geometry says that the layer is small, but in fact the image is still large

and if you do as you say -resize 
[output.zip](https://github.com/user-attachments/files/16502148/output.zip)

then rasterization will occur at 280x220
![image](https://github.com/user-attachments/assets/65984967-b98e-44d7-8235-0061840b4753)

and when you increase the layer to 1000x1000 you will see square pixels of bad quality =)
![image](https://github.com/user-attachments/assets/6a045998-a653-4ffd-92a3-ac2c009e774e)

just compare images of one by one of each 1000x1000

also in this file 
> magick identify -verbose "output.psd" > output.txt
> here is log [output.txt](https://github.com/user-attachments/files/16500477/output.txt)

there is no information about 280
so where is this data? i did -page 280x228+0+0 but in output the source resolution 836x709, 280 data even doesnt exist

**Reply by fmw42** [reply, marked as answer]
That is not what -page does, nor is it what to expect from most raster images. PSD may be different. I do not know and have never seen this behavior in any raster format or even PSD, unless the image dimensions are very large so that one can zoom in without loss of detail. I do not know photopeia nor what they are doing in their processing to create the PSD files. Is there documentation?
